src/ripple/app/ledger/OrderBookDB.cpp

```diff
@@ -272,12 +273,15 @@ void OrderBookDB::processTxn (
                             data->isFieldPresent (sfTakerGets))
                         {
                             // determine the OrderBook
-                            auto listeners = getBookListeners (
-                                {data->getFieldAmount (sfTakerGets).issue(),
-                                 data->getFieldAmount (sfTakerPays).issue()});
-
-                            if (listeners)
-                                listeners->publish (jvObj);
+                            Book b{data->getFieldAmount(sfTakerGets).issue(),
+                                data->getFieldAmount(sfTakerPays).issue()};
+
+                            auto listeners = getBookListeners(b);
+                            if (listeners &&
+                                havePublished.emplace(std::move(b)).second)
```

Comment:
The `std::move` doesn't do any harm, but books don't benefit from moving. I'd just `emplace(b)`. (This is a nit, your call if you want to keep the move).